fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/ContainmentIndexImpl.java

```diff
@@ -696,4 +712,38 @@ private Timestamp formatInstant(final Instant instant) {
         timestamp.setNanos(0);
         return timestamp;
     }
+
+    /**
+     * Private class to back a stream with a paged DB query.
+     *
+     * If this needs to be run in parallel we will have to override trySplit() and determine a good method to split on.
+     */
+    private class ContainmentIterator extends Spliterators.AbstractSpliterator<String> {
+        final Queue<String> children = new LinkedBlockingQueue<>();
+        final AtomicInteger numOffsets = new AtomicInteger(0);
+        final String queryToUse;
+        final MapSqlParameterSource parameterSource;
+
+        public ContainmentIterator(final String query, final MapSqlParameterSource parameters) {
+            super(Long.MAX_VALUE, Spliterator.ORDERED);
+            queryToUse = query;
+            parameterSource = parameters;
+        }
+
+        @Override
+        public boolean tryAdvance(final Consumer<? super String> action) {
+            try {
+                action.accept(children.remove());
+            } catch (final NoSuchElementException e) {
+                parameterSource.addValue("offSet", numOffsets.getAndIncrement() * containsLimit);
```

Comment:
Is there a reason to not set the `containsLimit` parameter here too?

I see, because there's no reason to set it every time. Perhaps set it in the constructor then?

Reply:
I was setting it in the calling method where I set all of the other parameters (ie. parent, startTime, child). I can move it into this method if that makes it clearer. But yes, LIMIT is a constant and we just advance the OFFSET parameter.